### ISSUE TITLE
Twig Template Loading Error - v2.0

### ISSUE THREAD

**Issue by jalendport** [focuslabllc/craft-cheat-sheet#54]
Using the latest version of Craft 3 and Craft Cheat Sheet.

Just downloaded it and attempting to visit `mysite.test/cheatsheet` results in the following error:

    Twig Template Loading Error – craft\web\twig\TemplateLoaderException
    Unable to find the template “_includes/fieldMacros.twig” in "cheatsheet" at line 2.

It's quite possible it's some misconfiguration or weird setup on my site so please don't hesitate to ask for log files or whatever to make debugging easier...

**Comment by erikreagan** [comment]
Hey Jalen. Did you install it via the plugin store? Another method?
-- 
Erik Reagan
Co-Founder, Director
Focus Lab, LLC
http://focuslabllc.com

**Comment by jalendport** [comment]
Hey! Via the plugin store.

**Comment by pogorzelskip** [comment]
Same here. I run latest Craft on xampp.

**Comment by erikreagan** [comment]
>It's quite possible it's some misconfiguration or weird setup on my site so please don't hesitate to ask for log files or whatever to make debugging easier...

That's an interesting statement. Did you install or set things up in a peculiar manner?

If it's okay I may end up hitting you up in Slack in the next day or so to speed up some back and forth just as a heads up. :)

**Comment by erikreagan** [comment]
@jalendport @pogorzels Can one (or both) of you share your `composer.json` and `composer.lock` files from the root of your project? I'd like to explore the overlaps and see if there may be a conflict with another plugin at play.

Feel free to share them here or email them to dev@focuslabllc.com.

**Comment by erikreagan** [comment]
@jalendport I just published 2.0.2 which I believe will resolve this. Let me know if that's not the case though. Thanks!

**Comment by DiederikVanHoorebeke** [comment]
I have v2.0.2 installed, and I get a similar error, but on lines 139-145. The plugin could not load `_includes/fieldMacros.twig` in `/craft/vendor/focuslabllc/craft-cheat-sheet/src/templates/frontEnd/cheatsheet.html`.

**Comment by erikreagan** [comment]
@DiederikVanHoorebeke We had an error in the version tagging that may have cause the update to not work correctly. Can you try updating the plugin via composer and see if that fixes the problem?

**Comment by erikreagan** [comment]
@DiederikVanHoorebeke Apologies. I forgot we hadn't published to Packagist yet. You should be able to use composer now and not receive that message.

**Comment by DiederikVanHoorebeke** [comment]
I'm afraid it didn't help...

I uninstalled the plugin in the CP and did the following:
```
$ composer clearcache
Cache directory does not exist (cache-vcs-dir):
Clearing cache (cache-repo-dir): ~/.composer/cache/repo
Clearing cache (cache-files-dir): ~/.composer/cache/files
Clearing cache (cache-dir): ~/.composer/cache
All caches cleared.
$ composer require focuslabllc/craft-cheat-sheet
Using version ^2.0 for focuslabllc/craft-cheat-sheet
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing focuslabllc/craft-cheat-sheet (2.0.2): Downloading (100%)
Writing lock file
Generating optimized autoload files
```

Then I installed and configured the plugin to get:
![image](https://user-images.githubusercontent.com/3625639/43768199-10ed1bee-9a37-11e8-850c-45eccf146142.png)

**Comment by MattWilcox** [comment]
I'm getting this error also, on a fresh install of Craft3 and a fresh install of this plugin. I don't use Composer at all; the Craft3 install was from the zip, if that makes any difference. The plugin install was through the admin panel and plugin store.

**Comment by erikreagan** [comment]
Thanks guys. Reopening as we investigate further.

**Comment by khalwat** [comment]
You can fix this by changing it to a full path here:


```
                     {{ view.renderTemplateMacro('cheat-sheet/frontEnd/_includes/fieldMacros.twig', csMacros[fieldClass], [
                            {
                                'sampleTag' : sampleTag ~ '.' ~ field.handle,
                                'field'     : field,
                                'whitespace': whitespace
                            }
                        ])|raw }}
```

Once a template starts rendering, there’s no guarantee that something else won’t be rendering something, so you can’t count on just changing the template path.

Also in the controller, you could make it a little more robust with some sanity checking/error logging:

```
    public function actionFieldCheatSheet()
    {
        // Only show the cheat sheet if we're in dev mode
        // Otherwise throw a 404 to the browser
        if (!YII_DEBUG) {
            throw new NotFoundHttpException();
        }

        $plugin = Plugin::getInstance();
        $view = Craft::$app->view;

        // Stash the old template mode, and set it AdminCP template mode
        $oldMode = Craft::$app->view->getTemplateMode();
        try {
            Craft::$app->view->setTemplateMode(View::TEMPLATE_MODE_CP);
        } catch (Exception $e) {
            Craft::error($e->getMessage(), __METHOD__);
        }
        $view->setTemplateMode(View::TEMPLATE_MODE_CP);
        $view->registerAssetBundle(CheatSheetAsset::class);

        $params = $plugin->fields->getTemplateData();
        $templatePath = 'frontEnd/cheatsheet';

        // Render the template with our vars passed in
        try {
            $htmlText = Craft::$app->view->renderTemplate('cheat-sheet/' . $templatePath, $params);
        } catch (\Exception $e) {
            $htmlText = Craft::t(
                'cheat-sheet',
                'Error rendering `{template}` -> {error}',
                ['template' => $templatePath, 'error' => $e->getMessage()]
            );
            Craft::error($htmlText, __METHOD__);
        }
        $view->setTemplateMode($oldMode);

        return $htmlText;
    }
```

**Comment by erikreagan** [comment]
Update: this error was encountered when a few other plugins were present in a Craft install. @khalwat came in clutch (above) with a simple fix. The patch has been released with a fix to this in 2.0.3.
